projects/packages/paypal-payments/src/legacy/class-simple-payments.php

```diff
@@ -0,0 +1,834 @@
+	 *
+	 * @param string $content - the content of the post.
+	 */
+	public function remove_auto_paragraph_from_product_description( $content ) {
+		if ( get_post_type() === self::$post_type_product ) {
+			remove_filter( 'the_content', 'wpautop' );
+		}
+
+		return $content;
+	}
+
+	/** Return the blog ID */
+	public function get_blog_id() {
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+			return get_current_blog_id();
+		}
+
+		return Jetpack_Options::get_option( 'id' );
+	}
+
+	/**
+	 * Used to check whether Simple Payments are enabled for given site.
+	 *
+	 * @return bool True if Simple Payments are enabled, false otherwise.
+	 */
+	public function is_enabled_jetpack_simple_payments() {
+		/**
+		 * Can be used by plugin authors to disable the conflicting output of Simple Payments.
+		 *
+		 * @since 6.3.0
+		 *
+		 * @param bool True if Simple Payments should be disabled, false otherwise.
+		 */
+		if ( apply_filters( 'jetpack_disable_simple_payments', false ) ) {
+			return false;
+		}
+
+		return ( ( defined( 'IS_WPCOM' ) && IS_WPCOM )
+			|| Jetpack::is_connection_ready() )
+			&&
+			Jetpack_Plan::supports( 'simple-payments' );
+	}
+
+	/**
+	 * Get a WP_Post representation of a product
+	 *
+	 * @param int $id The ID of the product.
+	 *
+	 * @return array|false|WP_Post
+	 */
+	private function get_product( $id ) {
+		if ( ! $id ) {
+			return false;
+		}
+
+		$product = get_post( $id );
+		if ( ! $product || is_wp_error( $product ) ) {
+			return false;
+		}
+		if ( $product->post_type !== self::$post_type_product || 'publish' !== $product->post_status ) {
+			return false;
+		}
+		return $product;
+	}
+
+	/**
+	 * Creates the content from a shortcode
+	 *
+	 * @param array $attrs Shortcode attributes.
+	 * @param mixed $content unused.
+	 *
+	 * @return string|void
+	 */
+	public function parse_shortcode( $attrs, $content = false ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+		if ( empty( $attrs['id'] ) ) {
+			return;
+		}
+		$product = $this->get_product( $attrs['id'] );
+		if ( ! $product ) {
+			return;
+		}
+
+		// We allow for overriding the presentation labels.
+		$data = shortcode_atts(
+			array(
+				'blog_id'     => $this->get_blog_id(),
+				'dom_id'      => uniqid( self::$css_classname_prefix . '-' . $product->ID . '_', true ),
+				'class'       => self::$css_classname_prefix . '-' . $product->ID,
+				'title'       => get_the_title( $product ),
+				'description' => $product->post_content,
+				'cta'         => get_post_meta( $product->ID, 'spay_cta', true ),
+				'multiple'    => get_post_meta( $product->ID, 'spay_multiple', true ) || '0',
+			),
+			$attrs
+		);
+
+		$data['price'] = $this->format_price(
+			get_post_meta( $product->ID, 'spay_price', true ),
+			get_post_meta( $product->ID, 'spay_currency', true )
+		);
+
+		$data['id'] = $attrs['id'];
+
+		if ( ! $this->is_enabled_jetpack_simple_payments() ) {
+			if ( jetpack_is_frontend() ) {
```

Comment:
this is a Jetpack-the-plugin only function. I think you'll need to move it to a package to start using it here.

Reply:
I've added a check in `get_instance()` for this function. If it doesn't exist it returns null. Added in [302ea6d](https://github.com/Automattic/jetpack/pull/43413/commits/302ea6d996a5ed98290ed5b87eda96a1e200028e).